tasties_app/tests/test_comment.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+from tasties_app.models import Comment, Recipe
+from django.contrib.auth.models import User
+from django.utils import timezone
+from datetime import date
+
+
+@pytest.fixture
+def generate_comments():
+    """
+    in this generate function we create User, Category and Recipe
+    objects to fill Comment class and used to test
+     """
+    user = User.objects.create_user(username='test_user', password='password')
+    recipe = Recipe(title='Test Recipe1', author_id=user, description='Test Description1',
+                    directions='Test Directions1', publication_date=timezone.now(), minutes_to_make=1,
+                    recipe_picture='recipe.jpeg')
+    recipe.save()
+    the_date = date.today().strftime("%Y-%m-%d")
+    comment = Comment(author_id=user, recipe_id=recipe, publication_date=the_date, comment_text='test comment')
+    comment.save()
+
+
+class TestComment():
+    @pytest.mark.django_db
+    def test_str(self, generate_comments):
+        user = User.objects.get(username='test_user')
+        comment = Comment.objects.filter(author_id=user)
+        the_date = str(date.today().strftime("%Y-%m-%d"))
+        expected_comment1_str = user.get_username()+' '+comment[0].comment_text+' '+the_date
```

Comment:
This makes a bad and incorrect assumption about the database, and can lead to issues down the road. Let's say that in the future, for some reason, we have a user with the same name `test_user`, and that he has in fact created many, many Comments. Then `Comment.objects.filter(author_id=user)[0]` will not be what you expect it to be here, and this test will cause massive issues.

I can suggest that your fixture `generate_comments` should return the Comment object that it saved to the database, and that instead of line 28 (`comment = Comment.objects.filter(author_id=user)`) you should have: `comment = Comment.objects.get(id=generate_comments.id)`, of course this should be in a `try-catch` block, since an `Exception `might get raised.

Reply:
i think its not work because if we have one more comment there will be problem `Comment.objects.get` return just one object.